code/modules/projectiles/updated_projectiles/guns/energy.dm

```diff
@@ -371,3 +371,235 @@
+//M43 Sunfury Lasgun MK1
+
+/obj/item/weapon/gun/energy/lasgun/M43
+	name = "\improper M43 Sunfury Lasgun MK1"
+	desc = "An accurate, recoilless laser based battle rifle with an integrated charge selector. Ideal for longer range engagements. Uses power cells."
+	force = 15 //Large and hefty!
+	icon_state = "m43"
+	item_state = "m43"
+	attachable_allowed = list(
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/verticalgrip,
+						/obj/item/attachable/angledgrip,
+						/obj/item/attachable/lasersight,
+						/obj/item/attachable/gyro,
+						/obj/item/attachable/flashlight,
+						/obj/item/attachable/bipod,
+						/obj/item/attachable/magnetic_harness,
+						/obj/item/attachable/attached_gun/grenade,
+						/obj/item/attachable/attached_gun/flamer,
+						/obj/item/attachable/attached_gun/shotgun,
+						/obj/item/attachable/scope,
+						/obj/item/attachable/scope/mini)
+
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER|GUN_ENERGY
+	starting_attachment_types = list(/obj/item/attachable/attached_gun/grenade)
+
+/obj/item/weapon/gun/energy/lasgun/M43/New()
+	..()
+	attachable_offset = list("muzzle_x" = 32, "muzzle_y" = 18,"rail_x" = 12, "rail_y" = 24, "under_x" = 23, "under_y" = 15, "stock_x" = 22, "stock_y" = 12)
+	var/obj/item/attachable/stock/lasgun/S = new(src)
+	S.flags_attach_features &= ~ATTACH_REMOVABLE
+	S.Attach(src)
+	update_attachables()
+	update_icon()
+	S.icon_state = initial(S.icon_state)
+
+
+/obj/item/weapon/gun/energy/lasgun/M43/set_gun_config_values()
+	fire_delay = config.low_fire_delay
+	accuracy_mult = config.base_hit_accuracy_mult + config.max_hit_accuracy_mult
+	accuracy_mult_unwielded = config.base_hit_accuracy_mult - config.max_hit_accuracy_mult //Heavy and unwieldy; you don't one hand this.
+	damage_mult = config.base_hit_damage_mult
+	scatter_unwielded = config.max_scatter_value * 2.5 //Heavy and unwieldy; you don't one hand this.
+	damage_falloff_mult = config.med_damage_falloff_mult
+
+//variant without ugl attachment
+/obj/item/weapon/gun/energy/lasgun/M43/stripped
+	starting_attachment_types = list()
+
+/obj/item/weapon/gun/energy/lasgun/M43/unique_action(mob/user)
+	toggle_chargemode(user)
+
+
+//Toggles Overcharge mode. Overcharge mode significantly increases damage and AP in exchange for doubled ammo usage and increased fire delay.
+/obj/item/weapon/gun/energy/lasgun/proc/toggle_chargemode(mob/user)
+	if(overcharge == 0)
+		if(current_mag.current_rounds < 1)
+			playsound(user, 'sound/machines/buzz-two.ogg', 15, 1)
+			to_chat(user, "<span class='warning'>You attempt to toggle on [src]'s overcharge mode but your battery pack lacks adequate charge to do so.</span>")
+			return
+		//While overcharge is active, double ammo consumption, and
+		playsound(user, 'sound/weapons/emitter.ogg', 15, 1)
+		ammo_per_shot = OVERCHARGE_AMMO_COST
+		fire_delay = config.med_fire_delay * 2 // 1 shot per second fire rate
```

Comment:
Higher, if it's going to be perfect-accuracy long-range 55 damage 55 AP.

Reply:
How much higher? Will you not be happy until the DPS is lower than a slug?

Reply:
For sure! A slug has much lower accuracy and can't be shot through a scope. Compare it to a sniper gun and bullet and make it weaker, much weaker.

Reply:
And it also knocks down, slows and disables, and is accurate enough to impose all of these effects consistently at the range it can do so.

You don't use slugs for the DPS, you use them for the spike damage and the debuffs.

Damage is the _only_ thing the overcharge has.

Reply:
Besides sniper range, full accuracy and a ton of AP. Unless this is a variant for the sniper spec, tweak it.

Reply:
AP that's only relevant vs a handful of xenos, and in practice unless you get a scope, 2 tiles more effective range.

At best the damage could use a minor tweak downward, but I'm not doubling the fire delay or doing whatever other crazy thing you had in mind.